src/Periphery/Permit2Proxy.sol

```diff
@@ -0,0 +1,356 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.17;
+
+import { ISignatureTransfer } from "permit2/interfaces/ISignatureTransfer.sol";
+import { TransferrableOwnership } from "lifi/Helpers/TransferrableOwnership.sol";
+import { LibAsset, IERC20 } from "lifi/Libraries/LibAsset.sol";
+import { PermitHash } from "permit2/libraries/PermitHash.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import { ERC20Permit } from "@openzeppelin/contracts/token/ERC20/extensions/ERC20Permit.sol";
+
+/// @title Permit2Proxy
+/// @author LI.FI (https://li.fi)
+/// @notice Proxy contract allowing gasless (Permit2-enabled) calls to our
+///         diamond contract
+/// @custom:version 1.0.0
+contract Permit2Proxy {
+    /// Storage ///
+
+    address public immutable LIFI_DIAMOND;
+    ISignatureTransfer public immutable PERMIT2;
+    mapping(address => bool) public diamondWhitelist;
+
+    string public constant WITNESS_TYPE_STRING =
+        "LIFICall witness)LIFICall(address tokenReceiver,address diamondAddress,bytes32 diamondCalldataHash)TokenPermissions(address token,uint256 amount)";
+    bytes32 public constant WITNESS_TYPEHASH =
+        keccak256(
+            "LIFICall(address tokenReceiver,address diamondAddress,bytes32 diamondCalldataHash)"
+        );
+    bytes32 public immutable PERMIT_WITH_WITNESS_TYPEHASH;
+
+    /// Types ///
+
+    // @dev LIFI Specific Witness to verify
+    struct LIFICall {
```

Comment:
I would prefer to name it LiFiCall to be more consistent with the rest of our codebase